### PR TITLE
Add new algorithm for SHELX76 constant wavelength format

### PR DESCRIPTION
**Pull request by rosswhitfield**
**Description of work.**

This algorithm will save a peaks workspace to a SHELX76 formatted file which is made to be compatible with Fullprof and WINGX. This is a simplified version compared to SaveHKL and it assumes a constant wavelength.

**To test:**
gunzip [peaks.nxs.gz](https://github.com/mantidproject/mantid/files/5691712/peaks.nxs.gz)

```python
peaks = LoadNexus('peaks.nxs')
SaveHKLCW(peaks, OutputFile='peaks.hkl', DirectionCosines=True)
```

the output should look like

```
Single crystal data
(3i4,2f8.2,i4,6f8.5)
1.00800  0   0
   0   0 -1023969.11  220.35   1-0.75750 0.75756-0.60035 0.60027 0.25646 0.25649
   0   0  -8  434.20   32.95   1-0.77022 0.77033-0.60387 0.60376 0.20516 0.20511
   0   1 -10  311.58   24.54   1-0.79355 0.79362-0.60495 0.41224 0.06560 0.44746
   0   1  -8  474.48   33.32   1-0.80037 0.80039-0.59874 0.40614-0.03049 0.44094
   1  -1  -9 2469.09   76.73   1-0.76064 0.56837-0.60132 0.79359 0.24461 0.21718
   1   0 -11  263.56   21.95   1-0.79578 0.60307-0.60402 0.60409 0.04352 0.52093
   1   0  -9  300.96   25.59   1-0.80098 0.60882-0.59537 0.59528-0.06296 0.52439
   1   0  -7  340.96   30.30   1-0.79462 0.60232-0.57501 0.57491-0.19478 0.55380
   1   1  -9 2935.60   75.39   1-0.78732 0.59466-0.56280 0.37039-0.25176 0.71358
   1   1  -7 1728.59   62.75   1-0.75424 0.56169-0.52035 0.32806-0.40044 0.75953
   1   1  -5 1128.82   55.43   1-0.68429 0.49187-0.44695 0.25416-0.57617 0.83275
   1   2  -9   79.24   11.66   1-0.77034 0.57806-0.53964 0.15491-0.33965 0.80116
   2  -1 -10  291.36   24.70   1-0.79320 0.40815-0.60506 0.79754 0.06878 0.44422
   2  -1  -8  269.67   26.19   1-0.80063 0.41571-0.59784 0.79020-0.03979 0.45030
   2   0 -10 1743.86   56.41   1-0.79456 0.40975-0.57491 0.57493-0.19533 0.70821
   2   0  -8 2561.42   73.46   1-0.77088 0.38595-0.54031 0.54025-0.33735 0.74778
   2   0  -6 9518.59  153.08   1-0.71795 0.33289-0.48089 0.48085-0.50329 0.81115
   2   0  -4 5937.85  130.53   1-0.61504 0.23014-0.38138 0.38155-0.69013 0.89524
   2   1  -8  256.36   21.45   1-0.72690 0.34198-0.49027 0.29773-0.48089 0.89129
   2   1  -6  349.06   26.42   1-0.65581 0.27082-0.41944 0.22709-0.62768 0.93546
   2   1  -4  438.80   31.11   1-0.54718 0.16266-0.32051 0.12808-0.77322 0.97833
   2   1  -2  698.18   40.62   1-0.39699 0.01222-0.19282 0.00024-0.89734 0.99993
   2   2  -4 1154.45   45.65   1-0.54066 0.15570-0.31478-0.06989-0.78013 0.98533
   2   2  -2  483.27   30.23   1-0.42092 0.03576-0.21266-0.17224-0.88182 0.98441
   2   2   030249.15  241.44   1-0.28383-0.10128-0.10102-0.28377-0.95354 0.95353
   3  -1  -9 3195.87   81.50   1-0.79984 0.22268-0.58736 0.77980-0.12352 0.58508
   3  -1  -7 1874.07   67.20   1-0.78487 0.20772-0.55914 0.75155-0.26710 0.62612
   3  -1  -5 1647.32   67.78   1-0.74180 0.16436-0.50635 0.69870-0.43970 0.69628
   3  -1  -3  366.75   33.90   1-0.64917 0.07178-0.41315 0.60550-0.63866 0.79260
   3  -1   1  162.61   22.47   1-0.26690-0.31059-0.08755 0.27992-0.95974 0.90839
   3   0  -7  266.57   22.43   1-0.71609 0.13887-0.47897 0.47875-0.50774 0.86690
   3   0  -5  416.26   29.31   1-0.63753 0.06026-0.40221 0.40219-0.65710 0.91357
   3   0  -3  597.22   36.30   1-0.52210-0.05495-0.29860 0.29860-0.79890 0.95280
   3   0  -1  347.79   28.11   1-0.37196-0.20552-0.17224 0.17238-0.91213 0.96335
   3   1  -3  302.61   23.17   1-0.47794-0.09925-0.26064 0.06822-0.83883 0.99272
   3   1  -1  296.97   23.22   1-0.35277-0.22426-0.15657-0.03582-0.92252 0.97387
```

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    def name(self):
+        return "SaveHKLCW"
+
+    def seeAlso(self):
+        return ["SaveHKL"]
+
+    def PyInit(self):
+        self.declareProperty(IPeaksWorkspaceProperty('Workspace', '', direction=Direction.Input),
+                             doc='PeaksWorkspace that should be used.')
```
Comment (Harrietbrown) [inline Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py:33]:
```suggestion
                             doc='PeaksWorkspace to be saved.')
```

File: Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        with open(outFile, 'w') as f:
+            if header:
+                f.write("Single crystal data\n")
+                if directionCosines:
+                    f.write("(3i4,2f8.2,i4,6f8.5)\n")
+                else:
+                    f.write("(3i4,2f8.2,i4)\n")
+                wavelengths = [p.getWavelength() for p in peak_ws]
+                print(np.std(wavelengths))
```
Comment (Harrietbrown) [inline Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py:64]: Is this print statement needed? this is what is causing your doctest to fail.

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
I noticed in the output file that these two values end up conected because of the formatting, I would assume that this would interfere with the program that is supposed to load this:
image

**Comment by rosswhitfield** [comment]
> I noticed in the output file that these two values end up conected because of the formatting, I would assume that this would interfere with the program that is supposed to load this:
> image

That is the exact format required and doesn't work otherwise, think Fortran format statements.

File: Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py (hunk 2 above)
Comment (rosswhitfield) [reply Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py]: I thought I had removed that, the doc-tests fail otherwise.

**Review by Harrietbrown — APPROVED** [review]
I have confirmed that the format of the output is correct. If all tests pass this will be good to go. good job.

**Pull request merged**